cpp/src/community/legacy/louvain.cu

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
Are you splitting the below louvain.cu file to this louvain.cu and the below louvain_impl.cuh? If it's the case, at least one of the new flies (preferably the longer one) should be created using git mv (instead of deleting and creating) to better preserve the history.

Reply:
I used git mv.  If I check out this branch and do a git log on the files it shows the history.  I believe this is a bug in the git web presentation.